src/pymor/discretizers/builtin/cg.py

```diff
@@ -1066,17 +1071,29 @@ def discretize_stationary_cg(analytical_problem, diameter=None, domain_discretiz
         BoundaryL2Functional = BoundaryL2ProductFunctional
 
     Li = [DiffusionOperator(grid, boundary_info, diffusion_constant=0, name='boundary_part')]
+    if energy_product:
+        eLi = [DiffusionOperator(grid, boundary_info, dirichlet_clear_columns=True, diffusion_constant=0)]
     coefficients = [1.]
 
     # diffusion part
     if isinstance(p.diffusion, LincombFunction):
+        scalar_diffusion = len(p.diffusion.shape_range) == 0 \
+                           or (len(p.diffusion.shape_range) == 1 and p.diffusion.shape_range[0] == 1)
         Li += [DiffusionOperator(grid, boundary_info, diffusion_function=df, dirichlet_clear_diag=True,
                                  name=f'diffusion_{i}')
                for i, df in enumerate(p.diffusion.functions)]
+        if energy_product:
+            eLi += [DiffusionOperator(grid, boundary_info, diffusion_function=p.diffusion, dirichlet_clear_diag=True,
+                                      dirichlet_clear_columns=True)]
         coefficients += list(p.diffusion.coefficients)
     elif p.diffusion is not None:
+        scalar_diffusion = np.array(p.diffusion).ndim == 0 \
+                or (np.array(p.diffusion).ndim == 1 and np.array(p.diffusion).shape[0] == 1)
```

Comment:
This code works whether or not `p.diffusion` is a `LincombFunction`. I would remove it from the if-clause and move it down to where it is needed.